sqlmesh/core/engine_adapter/bigquery.py

```diff
@@ -261,6 +276,63 @@ def fetchall(
         )
         return list(self._query_data)
 
+    def __add_nested_columns(
+        self,
+        alter_expressions: t.List[exp.Alter],
+    ) -> t.List[exp.Alter]:
+        """
+        Add nested columns for the given alter expressions using the BigQuery API directly.
```

Comment:
Took me a minute to figure out what does the return value suppose to represent. Perhaps worth adding in the docstring. 

Alternatively, it could be a good idea to split nested and non-nested expressions upstream.

Reply:
Right, factored it out to a function that splits the expressions beforehand